cpp/src/groupby/sort/group_quantiles.cu

```diff
@@ -55,55 +93,38 @@ struct quantiles_functor {
     //            so that sorting isn't required. Then add support for pre-sorted
 
     // prepare args to be used by lambda below
-    auto values_view     = column_device_view::create(values);
-    auto group_size_view = column_device_view::create(group_sizes);
-    auto result_view     = mutable_column_device_view::create(result->mutable_view());
+    auto values_view     = column_device_view::create(values, stream);
+    auto group_size_view = column_device_view::create(group_sizes, stream);
+    auto result_view     = mutable_column_device_view::create(result->mutable_view(), stream);
 
     // For each group, calculate quantile
-    thrust::for_each_n(rmm::exec_policy(stream)->on(stream),
-                       thrust::make_counting_iterator(0),
-                       num_groups,
-                       [d_values       = *values_view,
-                        d_group_size   = *group_size_view,
-                        d_result       = *result_view,
-                        d_group_offset = group_offsets.data().get(),
-                        d_quantiles    = quantile.data().get(),
-                        num_quantiles  = quantile.size(),
-                        interpolation] __device__(size_type i) {
-                         size_type segment_size = d_group_size.element<size_type>(i);
-
-                         auto selector = [i, &d_values, d_group_offset](size_type j) {
-                           return d_values.element<T>(d_group_offset[i] + j);
-                         };
-
-                         thrust::transform(thrust::seq,
-                                           d_quantiles,
-                                           d_quantiles + num_quantiles,
-                                           d_result.data<ResultType>() + i * num_quantiles,
-                                           [selector, segment_size, interpolation](auto q) {
-                                             return cudf::detail::select_quantile<ResultType>(
-                                               selector, segment_size, q, interpolation);
-                                           });
-
-                         for (size_t j = 0; j < num_quantiles; j++) {
-                           size_type group_size = d_group_size.element<size_type>(i);
-                           if (group_size == 0)
-                             d_result.set_null(i * num_quantiles + j);
-                           else
-                             d_result.set_valid(i * num_quantiles + j);
-                         }
```

Comment:
Converted this lambda in to functor `calculate_quantile_fn` to make it cleaner and easier to reuse.